common/src/main/java/tachyon/LeaderSelectorClient.java

```diff
@@ -62,11 +62,18 @@ public LeaderSelectorClient(String zookeeperAddress, String electionPath, String
 
     // Create a leader selector using the given path for management.
     // All participants in a given leader selection must use the same path.
-    // ExampleClient here is also a LeaderSelectorListener but this isn't required.
     CuratorFramework client =
         CuratorFrameworkFactory.newClient(mZookeeperAddress, new ExponentialBackoffRetry(
             Constants.SECOND_MS, 3));
     client.start();
+
+    // Sometimes, if the master crashes and restarts too quickly, zookeeper thinks the new client is
+    // still an old one. Close the "old" client and recreate a new one.
+    client.close();
```

Comment:
I think the idea is to pretend the newly opened client will be interpreted as the old client (when it might not be). Then, explicitly close the client which gets us to a clean state regardless of what might have happened prior. Maybe this should be refactored into a method to make more clear?